Lib/test/test_mimetypes.py

```diff
@@ -51,6 +51,12 @@ def test_non_standard_types(self):
         eq(self.db.guess_type('foo.xul', strict=False), ('text/xul', None))
         eq(self.db.guess_extension('image/jpg', strict=False), '.jpg')
 
+    def test_tricky_cases(self):
```

Comment:
Can we instead call it `test_filename_with_url_delimiters` to be more explicit ?

It would be nice to explain a bit why this is tricky, i.e., They would have different mime types if interpreted as URL as compared to when interpreted as filename because of the semicolon.